### PR TITLE
[VL] Decouple BUILD_BENCHMARK with BUILD_TEST

### PR DESCRIPTION
**Pull request by Yohahaha**
After this patch, user can run micro benchmark without enable BUILD_TEST.

### CODE REVIEW

**Comment by Yohahaha** [comment]
@marin-ma @PHILO-HE please help review this minor patch, thank you!

File: cpp/velox/CMakeLists.txt [hunk 1]
```diff
@@ -124,8 +124,6 @@ macro(ADD_VELOX_DEPENDENCIES)
   if(BUILD_TESTS)
     add_velox_dependency(exec::test "${VELOX_COMPONENTS_PATH}/exec/tests/utils/libvelox_exec_test_lib.a")
     add_velox_dependency(temp::path "${VELOX_COMPONENTS_PATH}/exec/tests/utils/libvelox_temp_path.a")
-  endif()
-  if(BUILD_TESTS OR BUILD_BENCHMARKS)
```
Comment (PHILO-HE) [inline cpp/velox/CMakeLists.txt:128]: Just verified building benchmarks doesn't depend on `libvelox_dwio_common_test_utils.a` indeed. Thanks for your fix!

**Review by PHILO-HE — APPROVED** [review]
@marin-ma, do you have any comment?

**Comment by marin-ma** [comment]
Looks good. Thanks for this cleanup!

**Pull request merged**